File: personal/src/main/java/com/easymi/personal/activity/MyPopularizeActivity.java
package com.easymi.personal.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.easymi.component.Config;
import com.easymi.component.base.RxBaseActivity;
import com.easymi.component.network.ApiManager;
import com.easymi.component.network.HaveErrSubscriberListener;
import com.easymi.component.network.HttpResultFunc2;
import com.easymi.component.network.MySubscriber;
import com.easymi.component.utils.CommonUtil;
import com.easymi.component.utils.EmUtil;
import com.easymi.component.utils.TimeUtil;
import com.easymi.component.utils.ToastUtil;
import com.easymi.personal.McService;
import com.easymi.personal.R;
import com.easymi.personal.entity.MyPopularizeBean;
import com.easymi.personal.entity.MyPopularizeMainBean;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class MyPopularizeActivity extends RxBaseActivity implements View.OnClickListener {

    private com.easymi.component.widget.CusToolbar myPopularizeCtb;
    private android.support.v7.widget.RecyclerView myPopularizeRv;
    private BaseQuickAdapter<MyPopularizeBean, BaseViewHolder> adapter;
    private android.widget.TextView myPopularizeHeaderTvCount;
    private android.widget.TextView myPopularizeHeaderTvMoney;
    private android.widget.TextView myPopularizeHeaderTvRecord;
    private android.widget.TextView myPopularizeHeaderTvCountOn;


    @Override
    public boolean isEnableSwipe() {
        return false;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_my_popularize;
    }

    @Override
    public void initToolBar() {
        super.initToolBar();
        myPopularizeCtb = findViewById(R.id.myPopularizeCtb);
        myPopularizeCtb.setTitle("推广详情").setLeftIcon(R.drawable.ic_arrow_back, new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    @Override
    public void initViews(Bundle savedInstanceState) {
        myPopularizeHeaderTvCount = findViewById(R.id.myPopularizeHeaderTvCount);
        myPopularizeHeaderTvMoney = findViewById(R.id.myPopularizeHeaderTvMoney);
        myPopularizeHeaderTvRecord = findViewById(R.id.myPopularizeHeaderTvRecord);
        myPopularizeHeaderTvRecord.setOnClickListener(this);
        myPopularizeHeaderTvCountOn = findViewById(R.id.myPopularizeHeaderTvCountOn);
        myPopularizeHeaderTvCountOn.setOnClickListener(this);

        myPopularizeRv = findViewById(R.id.myPopularizeRv);
        myPopularizeRv.setLayoutManager(new LinearLayoutManager(this));
        adapter = new BaseQuickAdapter<MyPopularizeBean, BaseViewHolder>(R.layout.item_my_popularize) {
            @Override
            protected void convert(BaseViewHolder helper, MyPopularizeBean item) {
                helper.setText(R.id.itemMyPopularizeTvName, item.passengerName).setText(R.id.itemMyPopularizeTvPhone, item.passengerPhone)
                        .setText(R.id.itemMyPopularizeTvTime, TimeUtil.getTime("yyyy-MM-dd", item.created * 1000)).setText(R.id.itemMyPopularizeTvMoney, CommonUtil.d2s(item.commissionAmount, "0.00"));
            }
        };
        adapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
                Intent intent = new Intent(MyPopularizeActivity.this, MyPopularizeFeeDetailActivity.class);
                intent.putExtra("data", (MyPopularizeBean) adapter.getData().get(position));
                startActivity(intent);
            }
        });

        myPopularizeRv.setAdapter(adapter);
        getData();
    }

    private void getData() {
        mRxManager.add(ApiManager.getInstance().createApi(Config.HOST, McService.class)
                .getPromoteAppList(EmUtil.getEmployInfo().phone)
                .map(new HttpResultFunc2<>())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new MySubscriber<MyPopularizeMainBean>(this, true, false, new HaveErrSubscriberListener<MyPopularizeMainBean>() {
                    @Override
                    public void onNext(MyPopularizeMainBean myPopularizeMainBean) {
                        myPopularizeHeaderTvCount.setText(myPopularizeMainBean.passengerNum + "人推广总计");
                        myPopularizeHeaderTvMoney.setText("￥" + CommonUtil.d2s(myPopularizeMainBean.commissionAmount, "0.00"));
                        adapter.setNewData(myPopularizeMainBean.promoterPassengerVos);
                        myPopularizeRv.setVisibility(myPopularizeMainBean.promoterPassengerVos.size() > 0 ? View.VISIBLE : View.GONE);
                    }

                    @Override
                    public void onError(int code) {
                        ToastUtil.showMessage(MyPopularizeActivity.this, "数据出现错误,请重试...");
                        finish();
                    }
                })));

    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent();
        if (v.getId() == R.id.myPopularizeHeaderTvRecord) {
            intent.setClass(this, MyPopularizeApplyRecordActivity.class);
        } else if (v.getId() == R.id.myPopularizeHeaderTvCountOn) {
            intent.setClass(this, MyPopularizeCountOnActivity.class);
        }
        startActivity(intent);
    }
}

File: personal/src/main/java/com/easymi/personal/activity/MyPopularizeActivity.kt
package com.easymi.personal.activity

import android.content.ComponentName
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.easymi.component.Config
import com.easymi.component.base.RxBaseActivity
import com.easymi.component.network.ApiManager
import com.easymi.component.network.HaveErrSubscriberListener
import com.easymi.component.network.HttpResultFunc2
import com.easymi.component.network.MySubscriber
import com.easymi.component.utils.CommonUtil
import com.easymi.component.utils.EmUtil
import com.easymi.component.utils.TimeUtil
import com.easymi.component.utils.ToastUtil
import com.easymi.personal.McService
import com.easymi.personal.R
import com.easymi.personal.entity.MyPopularizeBean
import com.easymi.personal.entity.MyPopularizeMainBean
import kotlinx.android.synthetic.main.activity_my_popularize.*
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class MyPopularizeActivity : RxBaseActivity(), View.OnClickListener {

    private val adapter: BaseQuickAdapter<MyPopularizeBean, BaseViewHolder> by lazy {
        object : BaseQuickAdapter<MyPopularizeBean, BaseViewHolder>(R.layout.item_my_popularize) {
            override fun convert(helper: BaseViewHolder?, item: MyPopularizeBean?) {
                helper?.run {
                    item?.run {
                        setText(R.id.itemMyPopularizeTvName, passengerName)
                        setText(R.id.itemMyPopularizeTvPhone, passengerPhone)
                        setText(R.id.itemMyPopularizeTvTime, TimeUtil.getTime("yyyy-MM-dd", created * 1000))
                        setText(R.id.itemMyPopularizeTvMoney, CommonUtil.d2s(commissionAmount, "0.00"))
                    }
                }
            }
        }
    };

    override fun isEnableSwipe() = false

    override fun getLayoutId() = R.layout.activity_my_popularize;

    override fun initToolBar() {
        myPopularizeCtb.setTitle("推广详情").setLeftIcon(R.drawable.ic_arrow_back) { finish() };
    }

    override fun initViews(savedInstanceState: Bundle?) {
        myPopularizeHeaderTvRecord.setOnClickListener(this);
        myPopularizeHeaderTvCountOn.setOnClickListener(this);
        myPopularizeRv.layoutManager = (LinearLayoutManager(this))
        adapter.setOnItemClickListener { adapter, view, position ->
            startActivity(Intent().apply {
                component = ComponentName(this@MyPopularizeActivity, MyPopularizeFeeDetailActivity::class.java)
                putExtra("data", adapter.data[position] as? MyPopularizeBean)
            })
        }
        myPopularizeRv.setAdapter(adapter)
        getData()
    }

    private fun getData() {
        ApiManager.getInstance().createApi(Config.HOST, McService::class.java)
                .getPromoteAppList(EmUtil.getEmployInfo().phone)
                .map(HttpResultFunc2<MyPopularizeMainBean>())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(MySubscriber<MyPopularizeMainBean>(this, true, false, object : HaveErrSubscriberListener<MyPopularizeMainBean> {
                    override fun onNext(t: MyPopularizeMainBean?) {
                        t?.run {
                            myPopularizeHeaderTvCount.text = ("${passengerNum}人推广总计");
                            myPopularizeHeaderTvMoney.text = "￥ ${CommonUtil.d2s(commissionAmount, "0.00")}";
                            adapter.setNewData(promoterPassengerVos);
                            myPopularizeRv.visibility = if (promoterPassengerVos.size > 0) View.VISIBLE else View.GONE;
                        }
                    }

                    override fun onError(code: Int) {
                        ToastUtil.showMessage(this@MyPopularizeActivity, "数据出现错误,请重试...");
                        finish();
                    }
                }))
    }

    override fun onClick(v: View?) {
        v?.run {
            startActivity(Intent().apply {
                component = ComponentName(this@MyPopularizeActivity,
                        if (id == R.id.myPopularizeHeaderTvRecord) MyPopularizeApplyRecordActivity::class.java
                        else MyPopularizeCountOnActivity::class.java)
            })
        }
    }
}